features/filesystem/littlefs/littlefs/lfs.c

```diff
@@ -724,6 +724,19 @@ static int lfs_dir_find(lfs_t *lfs, lfs_dir_t *dir,
         pathname += strspn(pathname, "/");
         pathlen = strcspn(pathname, "/");
 
+        // special case for root dir
+        if (pathname[0] == '\0') {
+            *entry = (lfs_entry_t){
+                .d.type = LFS_TYPE_DIR,
+                .d.elen = sizeof(entry->d) - 4,
+                .d.alen = 0,
+                .d.nlen = 0,
+                .d.u.dir[0] = lfs->root[0],
+                .d.u.dir[1] = lfs->root[1],
```

Comment:
Neat. Didn't know trailing commas were supported like this in all compilers.

Reply:
There are still a lot of reasons to use C over C++. While C++ has been adding a lot of high-level features, C has been adding a lot of quality-of-life improvements :)